moveit_core/collision_detection/src/world.cpp

```diff
@@ -228,11 +279,27 @@ bool World::moveObject(const std::string& object_id, const Eigen::Isometry3d& tr
   if (transform.isApprox(Eigen::Isometry3d::Identity()))
     return true;  // object already at correct location
   ensureUnique(it->second);
-  for (size_t i = 0, n = it->second->shapes_.size(); i < n; ++i)
-  {
-    ASSERT_ISOMETRY(transform)  // unsanitized input, could contain a non-isometry
-    it->second->shape_poses_[i] = transform * it->second->shape_poses_[i];
-  }
+  ASSERT_ISOMETRY(transform)  // unsanitized input, could contain a non-isometry
+  const Eigen::Isometry3d new_pose = transform * it->second->pose_;
+  setObjectPose(object_id, new_pose);
+
+  updateGlobalPoses_(it->second);
+  notify(it->second, MOVE_SHAPE);
+  return true;
+}
+
+bool World::moveObjectAbsolute(const std::string& object_id, const Eigen::Isometry3d& transform)
+{
+  auto it = objects_.find(object_id);
+  if (it == objects_.end())
+    return false;
+  if (transform.isApprox(Eigen::Isometry3d::Identity()))
```

Comment:
Shouldn't you compare it with the current transform of the object?
```C++
if (transform.isApprox(it->second)) return true;
```
Or I am misunderstanding what should be happening here?

Reply:
Hmmm, I see it is called **move**ObjectAbsolute. Then I would expect something like:
```C++
 setObjectPose(object_id, transform * it->second);
```
in the lines below.

Reply:
`it->second` is the object pointer, so `(transform.isApprox(it->second))` won't evaluate.

Also note the documentation for this function: 
> Move the object pose (thus moving all shapes and subframes in the object) to the given transform specified in world frame. The transform replaces the old pose.

Reply:
Ok, sorry, I forgot to read that.